### ISSUE TITLE
Throwing error doesn't work

### ISSUE THREAD

**Issue by piotr-pawlowski** [flash-oss/graphql-upload-minimal#37]
I have noticed that throwing error in a resolver that handles file upload gives no result. Request hangs forever and api client never gets a response. Max file size limit doesn't throw any exceptions when limit is reached. 

Similar questions:
https://github.com/meabed/graphql-upload-ts/issues/78
https://github.com/flash-oss/graphql-upload-minimal/issues/19

**Comment by koresar** [comment]
That sucks. 

Without uploads a request is parsed BEFORE hitting resolvers.

But when uploads present, the request is PAUSED and it's the developer's responsibility to consume the stream.

If a resolver throws, the "graphql-upload-minimal" module losses any control over the request. It's the GraphQL server implementation (like Apollo) responsibility to close the stream.

AFAIK, the original "graphql-upload" modules consumes the stream fro you, stores the files to the hard drive, and them streams from hard disk to your resolver(s). So, I'd recommend using the https://github.com/jaydenseric/graphql-upload instead of its forks (like this one).

**Comment by piotr-pawlowski** [comment]
I'd like to use original graphql-upload but I can't import it because of ESM module.

**Comment by koresar** [comment]
Oh, that's sucks twice.

Any ideas how we can solve this issue not only for you but for everyone else?

**Comment by uranderu** [comment]
Hi, I have this issue too and would love to help. But I'm afraid I don't have enough technological knowledge for this. However if I may suggest, maybe in the same way the original "graphql-upload" does?

**Comment by GitVitor** [comment]
Via CURL Request this problem didn't happen, this problem occurs only for Browser requests

**Comment by koresar** [comment]
@GitVitor did you try "copying the request as cURL" in the Network tab of Chrome?

**Comment by tlan16** [comment]
I had a similar issue today, the solution is to "consume" the request as early as possible before any other assertion. 

For example:
```typescript
import { MutationResolvers } from '@generated/graphql';
import { appDataSource } from '@database/data-source';
import { File } from '@database/models';
import { assertIsGraphQLUpload, readUploadedFile } from '@utilities/graphql/file-upload';
import { assertIsAllowedMimeType, uploadToS3 } from '@services/file/s3/upload-to-s3';
import { logger } from '@utilities/logger/logger';
import { fileToJson } from '@services/file';
import { assertContextHasAccount } from '@handlers/graphql/context';

export const fileUpload: MutationResolvers['fileUpload'] = async (_, { input }, context) => {
  assertContextHasAccount(context);
  logger.verbose('Start resolving fileUpload');

  if (!appDataSource.isInitialized) await appDataSource.initialize();

  const file = await input.file.promise;
  assertIsGraphQLUpload(file);
  /**
   * If input is a graphql upload, its content stream must be consumed. Otherwise it results in never ending request.
   */
  const fileBuffer = await readUploadedFile(file);
  const { filename: fileName, mimetype: mimeType } = file;
  assertIsAllowedMimeType(mimeType);

  const s3Link = await uploadToS3({
    fileBuffer,
    fileName,
    mimeType,
  });

  const databaseFile: File = new File();
  databaseFile.name = fileName;
  databaseFile.url = s3Link;
  databaseFile.createdBy = context.account;

  await appDataSource.getRepository(File).save(databaseFile);

  return await fileToJson(databaseFile);
};
```

if `const fileBuffer = await readUploadedFile(file);` is called after `assertIsAllowedMimeType(mimeType);` the request will hang, even when express's `response.send` is called.

**Comment by koresar** [comment]
I wrote my thoughts on the error here: https://github.com/jaydenseric/graphql-upload/issues/352#issuecomment-1401290104

I'd highly recommend reading it as an educational material.

**Comment by tlan16** [comment]
The node stream is by default paused unless something consumes it. I'm expecting the lib, likely graphql-upload either consumes the stream or pipe it to void as part of the error handling. But obviously doesn't fix when the resolver just returns void.

**Comment by koresar** [comment]
The lib pipes it:
1. https://github.com/flash-oss/graphql-upload-minimal/blob/3db84236ddc205e7b767ac76738db28981157af1/public/processRequest.js#L357
2. https://github.com/jaydenseric/graphql-upload/blob/e01b5d5541760d529b06c900883c5fa7febcff00/processRequest.mjs#L344

FYI

**Comment by Alex0007** [comment]
@koresar 
Hi! This change worked for me
https://github.com/meabed/graphql-upload-ts/pull/173/files

**Comment by koresar** [comment]
Hello all.

Good news! Seems like @Alex0007 has fixed it. :) The fix was published as v1.5.5
Here are the Release notes: https://github.com/flash-oss/graphql-upload-minimal/releases/tag/v1.5.5

Cause of the bug: the Express.js middleware was mishandling errors (connectivity or other types). Thus, the whole request was hanging. This is DOS an attack vector.

I'm going to close this issue, but feel free to chat further if any.

Slava Ukraini!
